doc/designs/evp_skey.md

```diff
@@ -0,0 +1,183 @@
+EVP_SKEY Design
+===============
+
+We currently presume that symmetric keys are represented as octet buffer. It
```

Comment:
Well it is not a presumption, it is what the API takes in, so perhaps:
The current API assume a byte buffer is always available and sufficient to represent a symmetric key.